presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveBucketing.java

```diff
@@ -176,9 +176,6 @@ static int getBucketHashCode(BucketingVersion bucketingVersion, List<TypeInfo> t
             return Optional.of(new HiveBucketFilter(ImmutableSet.of(singleBucket.getAsInt())));
         }
 
-        if (!effectivePredicate.getDomains().isPresent()) {
```

Comment:
It's not obvious that this is redundant (without reading `extractFixedValues`) and IntelliJ will still warn about the `get()` call. We could change the below code so that it works in either case
```java
Optional<Domain> domain = stream(effectivePredicate.getDomains())
        .flatMap(domains -> domains.entrySet().stream())